File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        d={
            ")":"(",
            "]":"[",
            "}":"{",
        }
        i=0
        result=[]
        while(i<len(s)):
            if s[i] in d.values():
                result.append(s[i])
            else:
                if result and result[-1]==d[s[i]]:
                    result.pop()
                else:
                    return False
            i+=1
        return len(result)==0

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();

        for(char i:s.toCharArray()){
            if(i=='(' || i=='{' || i=='['){
                st.push(i);
            }
            else if(st.size()!=0 && i==')' && st.peek()=='('){
                st.pop();
            }
            else if(st.size()!=0 && i=='}' && st.peek()=='{'){
                st.pop();
            }
            else if(st.size()!=0 && i==']' && st.peek()=='['){
                st.pop();
            }
            else{
                return false;
            }
        }
    return st.isEmpty();
    }
}